feature-find-meal/src/main/java/com/kinandcarta/feature/find/meal/adapter/MealsAdapter.kt

```diff
@@ -1,81 +1,75 @@
 package com.kinandcarta.feature.find.meal.adapter
 
 import android.view.LayoutInflater
-import android.view.View
 import android.view.ViewGroup
-import android.widget.Button
-import android.widget.TextView
 import androidx.core.content.ContextCompat
 import androidx.core.view.isVisible
 import androidx.recyclerview.widget.ListAdapter
 import androidx.recyclerview.widget.RecyclerView
 import com.kcc.kmmhackathon.shared.entity.Meal
 import com.kcc.kmmhackathon.shared.utility.DistanceUnit
 import com.kinandcarta.feature.find.meal.R
+import com.kinandcarta.feature.find.meal.databinding.MealItemRowBinding
 import com.kinandcarta.feature.find.meal.extension.getPortionsString
-import com.kinandcarta.feature.find.meal.viewmodel.Meals
 
 class MealsAdapter(
     private var distanceUnit: DistanceUnit = DistanceUnit.miles,
-    private val clickListener: (String, Int) -> Unit
+    private val clickListener: (String) -> Unit
 ) : ListAdapter<Meal, MealsAdapter.MealsViewHolder>(MealDiffCallback()) {
 
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
-        LayoutInflater.from(parent.context)
-            .inflate(R.layout.meal_item_row, parent, false)
-            .run(::MealsViewHolder)
-
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MealsViewHolder {
+       val itemBinding = MealItemRowBinding.inflate(
+           LayoutInflater.from(parent.context), parent, false)
+        return MealsViewHolder(itemBinding)
+    }
 
     override fun onBindViewHolder(holder: MealsViewHolder, position: Int) {
         val item = getItem(position)
         holder.bindData(item, position)
     }
 
-    inner class MealsViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
-        private val nameView = itemView.findViewById<TextView>(R.id.mealName)
-        private val tempView = itemView.findViewById<TextView>(R.id.mealTemp)
-        private val infoView = itemView.findViewById<TextView>(R.id.mealInfo)
-        private val distanceView = itemView.findViewById<TextView>(R.id.mealDistance)
-        private val availableView = itemView.findViewById<TextView>(R.id.mealAvailable)
-        private val expiryView = itemView.findViewById<TextView>(R.id.mealExpiry)
-        private val portionsView = itemView.findViewById<TextView>(R.id.mealPortions)
-        private val reserveButton = itemView.findViewById<Button>(R.id.reserveButton)
+    inner class MealsViewHolder(private val itemBinding: MealItemRowBinding): RecyclerView.ViewHolder(itemBinding.root) {
+        private val ctx = itemBinding.root.context
 
         fun bindData(meal: Meal, position: Int) {
```

Comment:
Minor: if you add `with(itemBinding)` you can stop repeating yourself in the body function.

```suggestion
        fun bindData(meal: Meal, position: Int) = with(itemBinding) {
```